enterprise/backend/src/metabase_enterprise/workspaces/api.clj

```diff
@@ -0,0 +1,86 @@
+(ns metabase-enterprise.workspaces.api
+  "`/api/ee/workspace/` routes"
+  (:require
+   [honey.sql.helpers :as sql.helpers]
+   [metabase-enterprise.workspaces.common :as ws.common]
+   [metabase.api.common :as api]
+   [metabase.api.macros :as api.macros]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.request.core :as request]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+;;; schemas
+
+(def CreateWorkspace
+  "Schema for creating a new workspace"
+  [:map
+   [:name [:string {:min 1}]]
+   [:graph {:optional true} [:map]]])
+
+(def UpdateWorkspace
+  "Schema for updating a workspace"
+  [:map
+   [:name {:optional true} [:string {:min 1}]]
+   [:graph {:optional true} [:map]]])
+
+(def Workspace
+  "Schema for workspace response"
+  [:map
+   [:id ms/PositiveInt]
+   [:name :string]
+   [:graph [:maybe [:map]]]
```

Comment:
The graph structure is likely to have some churn, and be quite complicated.

Let's keep it encapsulated, and expose projections of it only as needed (e.g. the listing of entities)